### PR TITLE
Clean up util/serialization.{cpp,h} and add unit tests

### PR DESCRIPTION
**Pull request by kwolekr**
This also cuts down on repetitious code which minimizes the surface area for mistakes.

### CODE REVIEW

File: src/unittest/test_serialization.cpp [hunk 1]
```diff
@@ -38,10 +38,14 @@ class TestSerialization : public TestBase {
...
 	void testDeSerializeWideString();
 	void testDeSerializeLongString();
+	void testStreamRead();
+	void testStreamWrite();
 
 	std::string teststring2;
 	std::wstring teststring2_w;
 	std::string teststring2_w_encoded;
+
+	static const u8 test_serialized_data[12 * 13];
```
Comment (est31) [inline src/unittest/test_serialization.cpp]: Does it have to be public?

File: src/util/serialize.h [hunk 2]
```diff
@@ -32,183 +32,144 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #define FIXEDPOINT_FACTOR 1000.0f
 #define FIXEDPOINT_INVFACTOR (1.0f/FIXEDPOINT_FACTOR)
 
+#define STRING_MAX_LEN 0xFFFF
+#define WIDE_STRING_MAX_LEN 0xFFFF
+// 64 MB ought to be enough for anybody - Billy G.
```
Comment (est31) [inline src/util/serialize.h]: should be [MiB](https://en.wikipedia.org/wiki/Template:Quantities_of_bytes) :)

**Comment by est31** [comment]
looks good.

File: src/unittest/test_serialization.cpp (hunk 1 above)
Comment (kwolekr) [inline src/unittest/test_serialization.cpp]: Why not?
It's an extra 8 characters to write; not worth the effort

File: src/util/serialize.h (hunk 2 above)
Comment (kwolekr) [inline src/util/serialize.h]: screw those new-age data size specifiers

**Comment by kwolekr** [comment]
https://github.com/minetest/minetest/commit/bf991bde45b279b40fdfa6c6b6fc6e4dd265c7ee

**Pull request closed without merging**